io/plugins/eu.esdihumboldt.hale.io.codelist.skos/src/eu/esdihumboldt/hale/io/codelist/skos/reader/SkosCodeListReader.java

```diff
@@ -0,0 +1,55 @@
+package eu.esdihumboldt.hale.io.codelist.skos.reader;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+
+import eu.esdihumboldt.hale.common.codelist.CodeList;
+import eu.esdihumboldt.hale.common.codelist.io.CodeListReader;
+import eu.esdihumboldt.hale.common.core.io.IOProviderConfigurationException;
+import eu.esdihumboldt.hale.common.core.io.ProgressIndicator;
+import eu.esdihumboldt.hale.common.core.io.impl.AbstractImportProvider;
+import eu.esdihumboldt.hale.common.core.io.report.IOReport;
+import eu.esdihumboldt.hale.common.core.io.report.IOReporter;
+
+/**
+ * Reads a SKOS code list from rdf file or url
+ * 
+ * @author Arun
+ */
+public class SkosCodeListReader extends AbstractImportProvider implements CodeListReader {
+
+	private CodeList codelist;
+
+	@Override
+	public boolean isCancelable() {
+		return false;
+	}
+
+	@Override
+	protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
+			throws IOProviderConfigurationException, IOException {
+		progress.begin("Loading SKOS code list", ProgressIndicator.UNKNOWN);
+		try {
+			URI loc = getSource().getLocation();
+			InputStream in = getSource().getInput();
+			codelist = new SkosCodeList(in, loc);
+			progress.setCurrentTask("Code list loaded.");
+		} catch (Exception ex) {
+			throw new RuntimeException(ex);
```

Comment:
You should handle the exception here, report the error and report the whole process as failed.